servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -399,7 +415,13 @@ private Single<HttpServerContext> doBind(final HttpExecutionContext executionCon
         ReadOnlyHttpServerConfig roConfig = config.asReadOnly();
         StreamingHttpService filteredService = applyInternalFilters(service, roConfig.lifecycleObserver());
 
-        if (roConfig.tcpConfig().isAlpnConfigured()) {
+        if (roConfig.tcpConfig().sslConfig() != null && roConfig.tcpConfig().acceptInsecureConnections()) {
+            config.tcpConfig().sslConfig(null, null);
```

Comment:
We should not modify user-provided config. Consider using `new HttpServerConfig(config)` copy ctor that you added and then modify and create `roConfigWithoutSsl`.